connectors/injected.ts

```diff
@@ -7,8 +7,20 @@ export default class Connector extends LockConnector {
       provider = window['ethereum'];
       try {
         await window['ethereum'].request({ method: 'eth_requestAccounts' })
-      } catch (e) {
+      } catch (e: any) {
         console.error(e);
+        if (e.message = "Already processing eth_requestAccounts. Please wait.") {
+          try {
+            await provider.request({
+              method: "wallet_requestPermissions",
+              params: [{ eth_accounts: {} }],
+            });
+          } catch (e: any) {
+            console.error(e);
+            if (e.code === 4001) return;
+          }
+        }
```

Comment:
Now i got new changes in my local. but If the user ignores the second request, then it is still stuck
<img width="861" alt="Untitled 7" src="https://github.com/user-attachments/assets/efa69cc1-d7ec-4097-8c0a-f47955cb282c">

Reply:
You should not seet the second error (with requestPermission).

How many pending requests you get on the metamask icon ?

Reply:
2

Reply:
mhhh... it should not get more than 1.

Did you log out first before locking metamask ?

Reply:
Yep. try this:
- Log out and lock metamask
- Try to log in
- Ignore the metamask unlock prompt
- The "Connect wallet" button should remain in a loading state
- Refresh the page
- Try to log in
- Ignore the metamask unlock prompt
- See error
- even on reload you will see same from now

Reply:
Did you try with sx repo on master ?

Reply:
Yep on master

Reply:
![Screenshot 2024-07-27 at 19 47 53](https://github.com/user-attachments/assets/70156df3-5e44-4dae-b073-2395e4196971)

You should get these error